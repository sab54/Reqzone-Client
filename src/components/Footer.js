/**
 * Footer.js
 *
 * This component renders a footer section for the application.
 *
 * Key functionalities:
 * - **Dynamic Theming**: Background and text colors are applied from the provided `theme` object.
 * - **Branding Text**: Displays the static message `"Designed with ❤️ by Su"`.
 *
 * Component Structure:
 * 1. **Container View**: Positioned with padding and centered alignment.
 * 2. **Text Element**: Styled with small font size, Poppins font family, and reduced opacity.
 *
 * Props:
 * - `theme`: An object containing color values.
 *    - `footerBackground`: Background color for the footer.
 *    - `text`: Color for the footer text.
 *
 * Notes:
 * - The `Poppins` font must be loaded in the app (e.g., via `expo-font`).
 * - This component is purely presentational and does not manage state.
 *
 * Author: Sunidhi Abhange
 */

import React from 'react';
import { Text, View, StyleSheet } from 'react-native';

const Footer = ({ theme }) => {
    return (
        <View
            style={[
                styles.container,
                { backgroundColor: theme.footerBackground },
            ]}
        >
            <Text style={[styles.text, { color: theme.text }]}>
                Designed with ❤️ by Su
            </Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
        paddingVertical: 20,
    },
    text: {
        fontSize: 13,
        fontFamily: 'Poppins',
        opacity: 0.6,
    },
});

export default Footer;
