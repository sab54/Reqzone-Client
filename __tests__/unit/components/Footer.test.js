/**
 * Footer.test.js
 *
 * What This Test File Covers:
 *
 * 1. Renders Footer text
 *    - Confirms that "Designed with ❤️ by Su" appears in the output.
 *
 * 2. Applies theme colors (without relying on testID or fragile parent traversal)
 *    - Asserts the Text color comes from the provided theme via query.
 *    - Asserts the root View backgroundColor comes from the theme via toJSON().
 *
 * 3. Snapshot
 *    - Captures stable render output for regression testing.
 *
 * Notes:
 * - Uses @testing-library/react-native for render and queries.
 * - Does NOT require any changes to the Footer component.
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import Footer from 'src/components/Footer';

const mockTheme = {
  footerBackground: '#123456',
  text: '#abcdef',
};

describe('Footer', () => {
  it('renders the footer text correctly', () => {
    const { getByText } = render(<Footer theme={mockTheme} />);
    expect(getByText('Designed with ❤️ by Su')).toBeTruthy();
  });

  it('applies theme background and text colors (robust root inspection)', () => {
    const { getByText, toJSON } = render(<Footer theme={mockTheme} />);

    // Assert Text color using query result (handles style arrays)
    const textNode = getByText('Designed with ❤️ by Su');
    expect(textNode.props.style).toEqual(
      expect.arrayContaining([expect.objectContaining({ color: mockTheme.text })])
    );

    // Assert container background using the rendered JSON tree (root is the View)
    const root = toJSON(); // ReactTestRendererJSON | ReactTestRendererJSON[] | null
    expect(root).toBeTruthy();

    // Handle potential array (shouldn't be for this component, but keep it safe)
    const rootNode = Array.isArray(root) ? root[0] : root;
    expect(rootNode.type).toBe('View');

    const containerStyle = rootNode.props.style;
    if (Array.isArray(containerStyle)) {
      expect(containerStyle).toEqual(
        expect.arrayContaining([expect.objectContaining({ backgroundColor: mockTheme.footerBackground })])
      );
    } else {
      expect(containerStyle).toEqual(
        expect.objectContaining({ backgroundColor: mockTheme.footerBackground })
      );
    }
  });

  it('matches snapshot', () => {
    const tree = render(<Footer theme={mockTheme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
