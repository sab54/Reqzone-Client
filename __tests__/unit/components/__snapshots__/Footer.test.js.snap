// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingVertical": 20,
      },
      {
        "backgroundColor": "#123456",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Poppins",
          "fontSize": 13,
          "opacity": 0.6,
        },
        {
          "color": "#abcdef",
        },
      ]
    }
  >
    Designed with ❤️ by Su
  </Text>
</View>
`;
